core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -512,6 +522,57 @@ private[spark] class TaskSetManager(
     }
   }
 
+  private[this] def sortPendingTasks(): Unit = {
+    val taskIndexs = (0 until numTasks).toArray
+    implicit def ord = new Ordering[Int] {
+      override def compare(x: Int, y: Int): Int =
+        getTaskInputSizeFromShuffledRDD(tasks(x)) compare
+          getTaskInputSizeFromShuffledRDD(tasks(y))
+    }
+    if (tasks.nonEmpty) {
+      // Sort the tasks based on their input size from ShuffledRDD.
+      pendingTasksForExecutor.foreach {
+        case (k, v) => pendingTasksForExecutor(k) = v.sorted
+      }
+      pendingTasksForHost.foreach {
+        case (k, v) => pendingTasksForHost(k) = v.sorted
+      }
+      pendingTasksForRack.foreach {
+        case (k, v) => pendingTasksForRack(k) = v.sorted
+      }
+      pendingTasksWithNoPrefs = pendingTasksWithNoPrefs.sorted
+      allPendingTasks = allPendingTasks.sorted
+    }
+  }
+
+  // Visible for testing
+  private[spark] def setTaskInputSizeFromShuffledRDD(inputSize: Map[Task[_], Long]) = {
+    taskInputSizeFromShuffledRDD.clear()
+    inputSize.foreach{
```

Comment:
I might miss something but is this not just adding all entries from one Map to another? does `++` do this directly?